ui/src/AudioRecorder.js

```diff
@@ -3,28 +3,45 @@ export class AudioRecorder {
     * Creates an AudioRecorder that can record utterances
```

Comment:
Just a heads up that there will probably be many merge conflicts with this file (I believe Eldrick & I both made changes) so let us know if you need any help merging all of the versions